### ISSUE TITLE
Preview feature feedback: Order by Relations support (`orderByRelation`)

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#5438]
Please share your feedback about the `orderByRelation` functionality that was released in [v2.16.0](https://github.com/prisma/prisma/releases/tag/2.16.0) in this issue.

- If you encounter a bug, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repo. 
- If the feature is working well for you, please share this in a comment below or leave a 👍 on this issue.

If you have any questions, don't hesitate to ask them in the [`#prisma-client`](https://prisma.slack.com/archives/CCWDULGUW) channel in the [Prisma Slack](https://slack.prisma.io).

**Comment by guerrerocarlos** [comment]
The feature is working great, but if it could not only count the number of relations, but also `sum`, `avg`, `min` and `max` some specific attribute of the relations, it would make it a killer feature.

For example, a very common use case: 

### Top 10 students with the best grades

```ts
await prisma.students.findMany({
  take: 10,
  orderBy: {
    exams: {
      avg: {
        points: 'desc' 
      }
    }
  },
});
```

**Comment by Brandejs** [comment]
Hi,

I prepare a simple project. There are 3 entities with composite keys. The last entity (TASK) has reference to Company and Project. I Can order tasks by ProjectName, but I cannot order them by CompanyName. 

`
prismaClientUnknownRequestError2 [PrismaClientUnknownRequestError]: 
Invalid `prisma.task.findMany()` invocation: Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState("42804"), message: "argument of JOIN/ON must be type boolean, not type record", detail: None, hint: None, position: Some(Original(383)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("parse_coerce.c"), line: Some(1156), routine: Some("coerce_to_boolean") }) }) })  at cb (/prisma-schema-orderby/node_modules/@prisma/client/runtime/index.js:35105:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  clientVersion: '2.23.0'
}`

[prisma-schema-orderby.zip](https://github.com/prisma/prisma/files/6551961/prisma-schema-orderby.zip)

**Comment by andresribeiro** [comment]
For me it doesn't work very well with skip. Not that this is a bug, but when the relation has the same quantity the order of the objects in the response may vary. I think this will only be solved when it becomes possible to sort by multiple fields

**Comment by 2color** [comment]
Closing this now that this has been released to General Availability in the [3.0.1 release](https://github.com/prisma/prisma/releases/tag/3.0.1)
